### PR TITLE
[docs] Carousel component jsdocs

### LINKED ISSUE #1729: [doc]: JSDocs for ProductImageCarousel components
**Describe the request**

Create or update comment blocks in for the Venia-UI [ProductImageCarousel][] component so that it adhere to [JSDocs][] standards.

[jsdocs]: https://devdocs.io/jsdoc/
[productimagecarousel]: https://github.com/magento/pwa-studio/tree/develop/packages/venia-ui/lib/components/ProductImageCarousel

**Possible solutions**

Add JSDoc comment blocks to the Carousel component in ProductImageCarousel.

See [ToastContainer][] for a good example of JSDoc comments.

[ToastContainer]: https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/ToastContainer/toastContainer.js

### PR DESCRIPTION
**Pull request by dani97**
## Description

Created JS Docs for Product Image Carousal Component

## Related Issue

Closes #1729 

### Verification Stakeholders

@jcalcaben 

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/components/ProductImageCarousel/carousel.js [hunk 1]
```diff
@@ -16,6 +16,16 @@ import Image from '../Image';
 const DEFAULT_IMAGE_WIDTH = 640;
 const DEFAULT_IMAGE_HEIGHT = 800;
 
+/**
+ * Carousel component for product images
```
Comment (jcalcaben) [inline packages/venia-ui/lib/components/ProductImageCarousel/carousel.js]: Can you provide a more detailed description for people not familiar with an image carousel?

File: packages/venia-ui/lib/components/ProductImageCarousel/carousel.js [hunk 2]
```diff
@@ -16,6 +16,16 @@ import Image from '../Image';
...
+ *
+ * @typedef ProductImageCarousel
+ * @kind functional component
+ *
+ * @param {props} props
+ *
+ *  * @returns {React.Element} React carousel component that displays product image .
```
Comment (jcalcaben) [inline packages/venia-ui/lib/components/ProductImageCarousel/carousel.js]:
```suggestion
 *  * @returns {React.Element} React carousel component that displays a product image.
```

File: packages/venia-ui/lib/components/ProductImageCarousel/carousel.js (hunk 1 above)
Comment (dani97) [reply packages/venia-ui/lib/components/ProductImageCarousel/carousel.js]: @jcalcaben added more description on carousel 👍

File: packages/venia-ui/lib/components/ProductImageCarousel/carousel.js (hunk 2 above)
Comment (dani97) [reply packages/venia-ui/lib/components/ProductImageCarousel/carousel.js]: done

File: packages/venia-ui/lib/components/ProductImageCarousel/thumbnail.js [hunk 3]
```diff
@@ -59,6 +70,20 @@ const Thumbnail = props => {
     );
 };
 
+/**
+ * Props for {@link Thumbnail}
+ *
+ * @typedef props
+ *
+ * @property {Object} classes An object containing the class names for the Thumnail component
```
Comment (jcalcaben) [inline packages/venia-ui/lib/components/ProductImageCarousel/thumbnail.js]:
```suggestion
 * @property {Object} classes An object containing the class names for the Thumbnail component
```

File: packages/venia-ui/lib/components/ProductImageCarousel/thumbnail.js [hunk 4]
```diff
@@ -59,6 +70,20 @@ const Thumbnail = props => {
...
+ * @property {string} classes.root classes for root container
+ * @property {string} classes.rootSelected classes for thumnail corresponding to selected item
```
Comment (jcalcaben) [inline packages/venia-ui/lib/components/ProductImageCarousel/thumbnail.js]:
```suggestion
 * @property {string} classes.rootSelected classes for the selected thumbnail item
```

**Comment by dani97** [comment]
> Just a few spelling mistakes on the thumbnail component to fix and this PR should be good

corrected

**Pull request merged**